example/examples/echo-server.rs

```diff
@@ -95,62 +95,49 @@ fn main() {
-                              client_addr, err);
-                    None
-                },
-            }
-        })
-
         .for_each(|(socket, client_addr)| {
-            // This closure is called whenever a new connection has been received and successfully
-            // upgraded to use secio/plaintext and echo.
-            println!("Successfully negotiated protocol with {}", client_addr);
-
-            // We loop forever in order to handle all the messages sent by the client.
-            let client_finished = {
-                let client_addr = client_addr.clone();
-                loop_fn(socket, move |socket| {
-                    let client_addr = client_addr.clone();
-                    socket.into_future()
-                        .map_err(|(err, _)| err)
-                        .and_then(move |(msg, rest)| {
-                            if let Some(msg) = msg {
-                                // One message has been received. We send it back to the client.
-                                println!("Received a message from {}: {:?}\n => Sending back \
-                                          identical message to remote", client_addr, msg);
-                                Box::new(rest.send(msg).map(|m| Loop::Continue(m)))
-                                    as Box<Future<Item = _, Error = _>>
-                            } else {
-                                // End of stream. Connection closed. Breaking the loop.
-                                println!("Received EOF from {}\n => Dropping connection",
-                                         client_addr);
-                                Box::new(Ok(Loop::Break(())).into_future())
-                                    as Box<Future<Item = _, Error = _>>
-                            }
-                        })
+            // This closure is called whenever a new connection has been received.
+            // `socket` is a future that will be triggered once the upgrade to secio, multiplex
+            // and echo is complete.
+            let client_addr = client_addr.to_string();
+            println!("Incoming connection from {}", client_addr);
+
+            socket
+                .and_then(move |socket| {
+                    println!("Successfully negotiated protocol with {}", client_addr);
+
+                    // We loop forever in order to handle all the messages sent by the client.
+                    loop_fn(socket, move |socket| {
+                        let client_addr = client_addr.clone();
+                        socket.into_future()
+                            .map_err(|(err, _)| err)
+                            .and_then(move |(msg, rest)| {
+                                if let Some(msg) = msg {
+                                    // One message has been received. We send it back to the client.
+                                    println!("Received a message from {}: {:?}\n => Sending back \
```

Comment:
Is this just for debugging now or should we bring in a logging library like the parity client, allowing users to flip some flags for extra debug data? Just a general question that I thought of when I saw this, doesn't have to be addressed in this PR obviously.

Reply:
This example is here to quickly demonstrate how to use libp2p, so I don't think a logging library is necessary.

Reply:
No definitely not, I was just talking generally, might be nice to be able to flip on something like trace messages, we can move that conversation to Riot :P